File: tests/ts/test_variable_flag.ts
/**
 * Test entry point for variable flag tests (TypeScript).
 */

import { MESSAGE_COUNT, getMessage, checkMessage } from './include/variable_flag_messages';
import { getMessageInfo } from '../generated/ts/serialization-test.structframe';
import { run } from './include/test_harness';

const TEST_NAME = 'VariableFlagMessages';
const PROFILES = 'bulk';

process.exit(run(MESSAGE_COUNT, getMessage, checkMessage, getMessageInfo, TEST_NAME, PROFILES));

File: tests/csharp/test_variable_flag.cs
/**
 * Test entry point for variable flag tests (C#).
 */

using System;
using StructFrame.SerializationTest;
using StructFrameTests;

class TestVariableFlag
{
    private const string TEST_NAME = "VariableFlagMessages";
    private const string PROFILES = "bulk";

    public static int Main(string[] args)
    {
        return TestHarness.Run(
            args,
            VariableFlagMessages.MESSAGE_COUNT,
            VariableFlagMessages.GetMessage,
            VariableFlagMessages.CheckMessage,
            (msgId) => MessageDefinitions.GetMessageInfo(msgId),
            TEST_NAME,
            PROFILES
        );
    }
}
